### DISCUSSION TITLE
Security

### DISCUSSION THREAD

**Discussion by shinokada** [msiemens/tinydb#367] [General]
How can I make it secure my json file?
If you know the json file name, it's possible to overwrite by others?

**Comment by msiemens** [comment]
Hey @shinokada, TinyDB's design goal is to be as simple as possible. As such, it doesn't provide a method or mechanism for encrypting or otherwise protecting the files it generates.

It really depends on the use case, how one would go about securing data stored by TinyDB. In general, there's two way of approaching this: 1) Use the operating system's file permission system (Linux: make it only readable/writable by the user that runs the program; Windows will have some analogue); or 2) write a [custom storage](https://tinydb.readthedocs.io/en/latest/extend.html#write-a-custom-storage) that encrypts/decrypts the data in some way before writing/reading it. 

However both of these approaches are beyond the scope of TinyDB itself. If you or someone else is interested in e.g. writing a blog post about this topic, that would be really welcome, but I don't have the time myself unfortunately.
